scripts/init.sh

```diff
@@ -1,184 +1,157 @@
+
+  sed -i "s/SCNODE_NET_NODENAME=.*/SCNODE_NET_NODENAME=${SCNODE_NET_NODENAME}/g" "${ENV_FILE}"
+
+  if [ "${role_value}" = "forger" ]; then
+    echo -e "\nYou are setting up a forger node with a zend node. Zend node may require a few hours to sync.\n"
+    read -rp "Do you want to download and use a seed file to speed up the process? ('yes' or 'no'): " use_seed_answer
+    while [[ ! "${use_seed_answer}" =~ ^(yes|no)$ ]]; do
+      echo -e "\nError: The only allowed answers are 'yes' or 'no'. Try again...\n"
+      read -rp "Do you want to download and use a seed file to speed up the process? ('yes' or 'no'): " use_seed_answer
     done
+    if [ "${use_seed_answer}" = "yes" ]; then
+      sed -i "s/ZEN_USE_SEED_FILE=.*/ZEN_USE_SEED_FILE=true/g" "${ENV_FILE}"
+    fi
   fi
 fi
 
-SCNODE_NET_NODENAME="ext-partner-$((RANDOM%100000+1))" || { echo "Error: could not set NODE_NAME variable for some reason. Fix it before proceeding any further.  Exiting..."; exit 1; }
-sed "${sed_i[@]}" "s/SCNODE_NET_NODENAME=.*/SCNODE_NET_NODENAME=${SCNODE_NET_NODENAME}/g" "${ROOT_DIR}/${ENV_FILE}"
-
-# Checking all the variables
-to_check=(
-  "SCNODE_CERT_SIGNERS_MAXPKS"
-  "SCNODE_CERT_SIGNERS_THRESHOLD"
-  "SCNODE_CERT_SIGNERS_PUBKEYS"
-  "SCNODE_CERT_MASTERS_PUBKEYS"
-  "SCNODE_ALLOWED_FORGERS"
-  "SCNODE_FORGER_RESTRICT"
-  "SCNODE_GENESIS_BLOCKHEX"
-  "SCNODE_GENESIS_SCID"
-  "SCNODE_GENESIS_POWDATA"
-  "SCNODE_GENESIS_MCBLOCKHEIGHT"
-  "SCNODE_GENESIS_COMMTREEHASH"
-  "SCNODE_GENESIS_WITHDRAWALEPOCHLENGTH"
-  "SCNODE_GENESIS_MCNETWORK"
-  "SCNODE_GENESIS_ISNONCEASING"
-  "SCNODE_NET_KNOWNPEERS"
-  "SCNODE_NET_MAGICBYTES"
-  "SCNODE_NET_NODENAME"
-  "SCNODE_NET_P2P_PORT"
-  "SCNODE_REST_PORT"
-  "SCNODE_USER_ID"
-  "SCNODE_GRP_ID"
-  "SCNODE_WS_SERVER_PORT"
-  "SCNODE_WS_SERVER_ENABLED"
-  "SCNODE_WALLET_SEED"
-  "SCNODE_WALLET_MAXTX_FEE"
-  "SC_COMMITTISH_EVMAPP"
-)
-
-# Making sure all the variables are set
-for var in "${to_check[@]}"; do
-  if [ -z "${!var:-}" ]; then
-    echo "Error: Environment variable ${var} is required."
-
-    # Unset all the sourced variables
-    for i in "${to_check[@]}"; do
-      unset "${i}"
-    done
-    sleep 5
+# shellcheck source=../deployments/eon/forger/.env
+source "${ENV_FILE}" || fn_die "Error: could not source ${ENV_FILE} file. Fix it before proceeding any further.  Exiting..."
+
+check_required_variables
+
+echo -e "\n\033[1m=== Creating symlink to compose file ===\033[0m"
+COMPOSE_FILE="${ROOT_DIR}/compose_files/docker-compose-${role_value}.yml"
+SYMLINK_COMPOSE_FILE="${DEPLOYMENT_DIR}/docker-compose.yml"
+if ! [ -L "${SYMLINK_COMPOSE_FILE}" ]; then
+  ln -sf "${COMPOSE_FILE}" "${SYMLINK_COMPOSE_FILE}"
+fi
+
+if [ "${role_value}" = "forger" ]; then
 
-    echo "Thanks for your interest in EON, your configuration is incomplete. Application will not be able to start. Please contact Horizen to fix this issue. Exiting ..."
-    exit 1
+  echo -e "\n\033[1m=== Creating symlink to seed file ===\033[0m"
```

Comment:
I would drop this message completely in case people will not be choosing `seed` method but seeing this message can be misleading.

If you keep it then change `seed file` to `seed.sh script file`